src/server/lib/okta/idx/introspect.ts

```diff
@@ -0,0 +1,143 @@
+import { joinUrl } from '@guardian/libs';
+import { InteractResponse } from './interact';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+import { logger } from '@/server/lib/serverSideLogger';
+import { z } from 'zod';
+import { OAuthError } from '@/server/models/okta/Error';
+import { trackMetric } from '../../trackMetric';
+
+const { okta } = getConfiguration();
+
+// Schema for the 'redirect-idp' object inside the introspect response remediation object
+export const redirectIdpSchema = z.object({
+	name: z.literal('redirect-idp'),
+	type: z.enum(['APPLE', 'GOOGLE']),
+	href: z.string(),
+	method: z.literal('GET'),
+});
+
+// Schema for the introspect response
+const introspectResponseSchema = z.object({
+	stateHandle: z.string(),
+	expiresAt: z.coerce.date(),
+	remediation: z.object({
+		type: z.string(),
+		value: z.array(
+			// social idp object
+			z.union([
+				redirectIdpSchema,
+				z.object({
+					// any other name thats not one of the above
+					name: z.string(),
+					type: z.string().optional(),
+					href: z.string().optional(),
+					method: z.string().optional(),
+				}),
+			]),
+		),
+	}),
+});
+export type IntrospectResponse = z.infer<typeof introspectResponseSchema>;
+
+const introspectResponseErrorSchema = z.object({
+	version: z.literal('1.0.0'),
```

Comment:
What happens with Zod if they bump the version - nothing interesting? Still parses? Does this really need to be a literal?

Reply:
It will fail to parse I think, as it's expecting a literal string, so anything else will fail.

Maybe I should instead look for a string that starts with `1.`?

This is assuming they're even semver for the api

Reply:
Maybe I'll add a log if it doesn't match `1.0.0` but will still let the validation pass.